src/runtime/registry/google-tag-manager.ts

```diff
@@ -1,36 +1,32 @@
 // WARNING: This file is automatically generated, do not manually modify.
 import { withQuery } from 'ufo'
 import type { GoogleTagManagerApi } from 'third-party-capital'
+import { defu } from 'defu'
 import { useRegistryScript } from '#nuxt-scripts-utils'
 import type { RegistryScriptInput } from '#nuxt-scripts'
-import { object, string } from '#nuxt-scripts-validator'
+import { object, string, optional } from '#nuxt-scripts-validator'
 
-export const GoogleTagManagerOptions = object({ id: string() })
+export const GoogleTagManagerOptions = object({ id: string(), dataLayerName: optional(string()) })
 
 declare global {
   interface Window extends GoogleTagManagerApi {}
 }
 export type GoogleTagManagerInput = RegistryScriptInput<typeof GoogleTagManagerOptions>
 
 export function useScriptGoogleTagManager<T extends GoogleTagManagerApi>(_options?: GoogleTagManagerInput) {
+  _options = defu(_options, { dataLayerName: 'defaultGtm' })
   return useRegistryScript<T, typeof GoogleTagManagerOptions>(_options?.key || 'google-tag-manager', options => ({
     scriptInput: {
       src: withQuery('https://www.googletagmanager.com/gtm.js', { id: options?.id }),
```

Comment:
the src needs to include the dataLayer name otherwise how will it link it :thinking:

Reply:
hmmm  it's not in TPC :eyes: . Is there a default name for data layers ? 
cc @flashdesignory

Reply:
Oh wait I just merged this and I forgot I left this comment.

Can we do some testing on it and make sure it's actually working? I'm pretty sure it will just use `dataLayer` if not configured.

cc: @flashdesignory